src/java.desktop/share/classes/sun/swing/MnemonicHandler.java

```diff
@@ -65,34 +53,27 @@ public static boolean isMnemonicHidden() {
         return isMnemonicHidden;
     }
 
-    static class AltProcessor implements KeyEventPostProcessor {
-        public boolean postProcessKeyEvent(final KeyEvent ev) {
-            if (ev.getKeyCode() != KeyEvent.VK_ALT) {
-                return false;
-            }
-
-            final JRootPane root = SwingUtilities.getRootPane(ev.getComponent());
-            final Window winAncestor = (root == null ? null : SwingUtilities.getWindowAncestor(root));
-
-            switch(ev.getID()) {
-                case KeyEvent.KEY_PRESSED:
-                    setMnemonicHidden(false);
-                    break;
-                case KeyEvent.KEY_RELEASED:
-                    setMnemonicHidden(true);
-                    break;
-            }
-
-            repaintMnemonicsInWindow(winAncestor);
-
-            return false;
+    /**
+     * Sets the state of the hide mnemonic flag. This flag is used by the
+     * component UI delegates to determine if the mnemonic should be rendered.
+     * This method is a non operation if the underlying operating system
+     * does not support the mnemonic hiding feature.
+     *
+     * @param hide true if mnemonics should be hidden
+     */
+    public static void setMnemonicHidden(final boolean hide) {
+        if (UIManager.getBoolean("Button.showMnemonics")) {
+            // Do not hide mnemonics if the UI defaults do not support this
+            isMnemonicHidden = false;
+        } else {
+            isMnemonicHidden = hide;
         }
     }
 
     /*
      * Repaints all the components with the mnemonics in the given window and all its owned windows.
      */
```

Comment:
I suggest converting the comment to javadoc comment:
```suggestion
    /**
     * Repaints all the components with the mnemonics in the given window and all its owned windows.
     */
```